### PR TITLE
Cleanup and update API

### PR DESCRIPTION
**Pull request by michaelfig**
Implement the most recent decisions as per the proposal.

### CODE REVIEW

**Review by jfparadis — APPROVED** [review]

File: src/E.js [hunk 1]
```diff
@@ -57,103 +52,22 @@ function EProxyHandler(x, HandledPromise) {
...
-          wrap(HandledPromise.applyMethod(x, prop, args)),
-        );
-      },
-      apply(_target, _thisArg, args = []) {
-        return wrap(HandledPromise.applyFunction(x, args));
+        return HandledPromise.get(x, prop);
       },
     });
 
   E.G = o => makeEGetterProxy(o);
```
Comment (erights) [inline src/E.js]: Is there a reason not to do
```
E.G = makeEGetterProxy;
E.resolve = HandledPromise.resolve;
```
?

File: src/index.d.ts [hunk 2]
```diff
@@ -74,16 +40,18 @@ interface EProxy {
...
    */
-  (x: unknown): ESingleMethod<Promise<unknown>>;
-  sendOnly: (x: unknown) => ESingleMethod<void>;
+  (x: unknown): ESingleMethod;
+  sendOnly: (x: unknown) => ESingleMethod;
   /**
-   * E.C(x) returns a chain where operations are selected by
-   * uppercase single-letter selectors.
+   * E.G(x) returns a proxy on which you can get arbitrary properties.
+   * Each of these properties returns a promise.  The property will be
```
Comment (erights) [inline src/index.d.ts]: The *promise* will...

File: src/index.js [hunk 3]
```diff
@@ -8,94 +8,39 @@ import makeE from './E';
...
 // ...
 
-// TODO: Maybe rename the global HandledPromise to something only the tildot rewriter uses.
-if (!globalThis.HandledPromise) {
-  /* eslint-disable no-use-before-define */
-  // Install the shim as best we can.
-  maybeExtendPromise(Promise);
-  globalThis.HandledPromise = makeHandledPromise(Promise);
-  /* eslint-enable no-use-before-define */
+// eslint-disable-next-line import/no-mutable-exports
```
Comment (erights) [inline src/index.js]: Do you need `hp` to be a live binding? If not, I'd prefer

```js
const hp = typeof HandledPromise === 'undefined' ? makeHandledPromise() : HandledPromise;
```
or better
```js
const hp = typeof HandledPromise !== 'undefined' ?  HandledPromise : makeHandledPromise();
```

Then you could also get right of the disabling of this warning against live bindings.

**Review by erights — APPROVED** [review]
LGTM

**Comment by warner** [comment]
note: this PR needs to be migrated to the new monorepo in `agoric-sdk`

**Comment by michaelfig** [comment]
Closing due to staleness.  Moved a while ago to https://github.com/endojs/endo/tree/HEAD/packages/eventual-send

**Pull request closed without merging**